src/controllers/customer.js

```diff
@@ -35,14 +35,48 @@ const createCustomer = async (req, res) => {
     // HTTP error codes: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses
     if (e instanceof PrismaClientKnownRequestError) {
       if (e.code === "P2002") {
-        return res.status(409).json({ error: "A customer with the provided email already exists" })
+        return res
+          .status(409)
+          .json({ error: "A customer with the provided email already exists" });
       }
     }
 
-    res.status(500).json({ error: e.message })
+    res.status(500).json({ error: e.message });
   }
-}
+};
+
+const updateCustomer = async (req, res) => {
+  const { id } = req.params;
+  const { name, contact } = req.body;
+
+  if (!name && !contact) {
+    return res.status(400).json({ error: "Missing field in request body" });
+  }
+
+  try {
+    const existingCustomer = await findCustomerDB(id);
+    if (!existingCustomer) {
+      return res
+        .status(404)
+        .json({ error: "Customer with that id does not exist" });
+    }
+
+    existingCustomer.name = name || existingCustomer.name;
+
+    if (contact) {
+      existingCustomer.contact = {
+        phone: contact.phone || existingCustomer.contact?.phone,
+        email: contact.email || existingCustomer.contact?.email,
+      };
+    }
+    const updatedCustomer = await updateCustomerDB(id, existingCustomer);
+    res.status(201).json({ customer: updatedCustomer });
+  } catch (error) {
+    console.log(error.message);
+  }
```

Comment:
I would look at reviewing this try/catch block. Currently if any errors do happen the user gets no feedback from their request and will end up with a timeout. At the very least you should consider adding something along the lines of `res.status(500).json({error: "server error"})`. I would have expected the 404 res to also be part of the error handling rather than in the main `try` block.